openflow15/match.go

```diff
@@ -411,17 +411,18 @@ func DecodeMatchField(class uint16, field uint8, length uint8, hasMask bool, dat
 		case OXM_FIELD_TCP_FLAGS:
 			val = new(TcpFlagsField)
 		default:
-			klog.Warningf("Unhandled Field: %d in Class: %d", field, class)
+			return nil, fmt.Errorf("Unhandled Field: %d in Class: %d", field, class)
 		}
 
 		if val == nil {
-			klog.Errorf("Bad pkt class: %v field: %v data: %v", class, field, data)
-			return nil, fmt.Errorf("Bad pkt class: %v field: %v data: %v", class, field, data)
+			err := fmt.Errorf("Bad pkt class: %v field: %v", class, field)
```

Comment:
This should not happen, since the only `val==nil` is the default case, but it is already returned in line 414?

Reply:
You are right, thanks, will fix in next commit.